### PR TITLE
Client: MediaPipelineCapabilities

### PR DESCRIPTION
**Pull request by skyshaha1**
Summary: Test the MediaPipelineCapabilities APIs in the client component
tests.
Type: Feature
Test Plan: Unittests, Component Tests.
Jira: RIALTO-440

### CODE REVIEW

File: tests/common/matchers/MediaKeysProtoRequestMatchers.h [hunk 1]
```diff
@@ -191,4 +192,17 @@ MATCHER_P(getSupportedKeySystemVersionRequestMatcher, keySystem, "")
     return (kRequest->key_system() == keySystem);
 }
 
+MATCHER_P(getSupportedMimeTypesRequestMatcher, sourceType, "")
```
Comment (skywilliamsonl) [inline tests/common/matchers/MediaKeysProtoRequestMatchers.h:195]: These should be in MediaPipeline matcher header.

File: tests/componenttests/client/mocks/MediaPipelineCapabilitiesModuleMock.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MEDIA_PIPELINE_CAPABILITIES_MODULE_MOCK_H_
+#define MEDIA_PIPELINE_CAPABILITIES_MODULE_MOCK_H_
+
+#include "MediaKeysProtoUtils.h"
```
Comment (skywilliamsonl) [inline tests/componenttests/client/mocks/MediaPipelineCapabilitiesModuleMock.h:23]: Needed?

File: tests/componenttests/client/stubs/MediaPipelineCapabilitiesModuleStub.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "MediaPipelineCapabilitiesModuleStub.h"
+#include "MediaKeysProtoUtils.h"
```
Comment (skywilliamsonl) [inline tests/componenttests/client/stubs/MediaPipelineCapabilitiesModuleStub.cpp:21]: I think most of these headers aren't needed

File: tests/componenttests/client/stubs/MediaPipelineCapabilitiesModuleStub.h [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef FIREBOLT_RIALTO_CLIENT_CT_MEDIA_PIPELINE_CAPABILITIES_MODULE_STUB_H_
+#define FIREBOLT_RIALTO_CLIENT_CT_MEDIA_PIPELINE_CAPABILITIES_MODULE_STUB_H_
+
+#include "IIpcServer.h"
+#include "MediaCommon.h"
```
Comment (skywilliamsonl) [inline tests/componenttests/client/stubs/MediaPipelineCapabilitiesModuleStub.h:23]: I think most of these headers aren't needed

File: tests/componenttests/client/tests/base/MediaKeysTestMethods.cpp
Comment (skywilliamsonl) [inline tests/componenttests/client/tests/base/MediaKeysTestMethods.cpp:1]: All new code should be in MediaPipeline methods

File: tests/componenttests/client/tests/base/MediaKeysTestMethods.cpp [hunk 5]
```diff
@@ -60,14 +60,23 @@ const std::string kInvalidKeySystem{"notExpected"};
 const std::vector<std::string> kKeySystems(firebolt::rialto::server::kSupportedKeySystems.begin(),
                                            firebolt::rialto::server::kSupportedKeySystems.end());
 const std::string kVersion{"123"};
+constexpr int32_t kMediaSourceTypeUnknown{0};
```
Comment (skywilliamsonl) [inline tests/componenttests/client/tests/base/MediaKeysTestMethods.cpp:63]: Should use enums for  kMediaSourceTypeUnknown, kMediaSourceTypeAudio & kMediaSourceTypeVideo (probably dont need to use variables for them, just use the enums)

File: tests/componenttests/client/tests/base/MediaKeysTestMethods.cpp [hunk 6]
```diff
@@ -660,4 +681,79 @@ void MediaKeysTestMethods::doesNotGetSupportedKeySystemVersion()
...
+    EXPECT_CALL(*m_mediaPipelineCapabilitiesModuleMock,
+                isMimeTypeSupported(_, isMimeTypeSupportedRequestMatcher(kMimeType), _, _))
+        .WillOnce(DoAll(SetArgPointee<2>(m_mediaPipelineCapabilitiesModuleMock->isMimeTypeSupportedResponse(true)),
+                        WithArgs<0, 3>(Invoke(&(*m_mediaPipelineCapabilitiesModuleMock),
+                                              &MediaPipelineCapabilitiesModuleMock::defaultReturn))));
+}
+
+void MediaKeysTestMethods::isMimeTypeSupported()
+{
+    EXPECT_EQ(m_mediaPipelineCapabilities->isMimeTypeSupported(kMimeType), true);
```
Comment (skywilliamsonl) [inline tests/componenttests/client/tests/base/MediaKeysTestMethods.cpp:743]: Can remove kMimeType and just use kVideoMimeType[0]

File: tests/componenttests/client/tests/base/MediaKeysTestMethods.h
Comment (skywilliamsonl) [inline tests/componenttests/client/tests/base/MediaKeysTestMethods.h:1]: Move to MediaPipeline methods

**Review by skyproudmanp — APPROVED** [review]

**Review by skyproudmanp — APPROVED** [review]

**Review by skywilliamsonl — APPROVED** [review]

**Pull request merged**